### PR TITLE
Feature/aqm plot fix

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

Fixed plots so that the data counts on top of the plots are not so crowded, plus ensured that they are integers.
General cleanup
Modified some parm files for EE2 check (removing verbosity, etc)

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Test the prep code.  

Then bring in prep from the auth repo and test the stats code.  You may use cyc=04 or 11 for your test.

Then bring in stats from the auth repo and test the plot code.  Let me see the plot in question.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Thanks for submitting this PR! One comment for you and @ShelleyMelchior-NOAA : 

I see that you removed the entire line that defined [OUTPUTROOT in parm/evs_config/aqm/config.evs.aqm.prod](https://github.com/NOAA-EMC/EVS/pull/232/files#diff-e742d6076f683ee45a28c4eefceb1478a000fc596b0f21239299800bbac12f0aL30). That line might actually be needed, it just shouldn't be a hardcoded path. Should still define OUTPUTROOT, just not with a hardcoded path?

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Additionally, I don't think the following two aqm updates were made in this PR yet. Can you add them, please? Thank you!

Remove verbosity, etc. on L14-15 in: [parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_daily_obsAIRNOW.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_daily_obsAIRNOW.conf#L14-L15)
Remove verbosity, etc. on L14-15 in:
[parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_hourly_obsAIRNOW.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_hourly_obsAIRNOW.conf#L14-L15)

**Comment by PerryShafran-NOAA** [comment]
I don't use it anywhere in the scripts, I believe.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA OK, if you aren't using it, they you can remove it. Also [consider removing line 29 above it that describes that line](https://github.com/NOAA-EMC/EVS/pull/232/files#diff-e742d6076f683ee45a28c4eefceb1478a000fc596b0f21239299800bbac12f0aL29). 

Additionally, I don't think the following two aqm updates were made in this PR yet. Can you add them, please? Thank you!

Remove verbosity, etc. on L14-15 in: [parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_daily_obsAIRNOW.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_daily_obsAIRNOW.conf#L14-L15)
Remove verbosity, etc. on L14-15 in:
[parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_hourly_obsAIRNOW.conf](https://github.com/NOAA-EMC/EVS/blob/6ba22d2551555704da0de4397128343d260befc3/parm/metplus_config/aqm/grid2obs/prep/Ascii2Nc_hourly_obsAIRNOW.conf#L14-L15)

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Thanks for making those other edits!

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Can you take this opportunity to also remove your DCOMROOT references in aqm like Shannon did for subseasonal? For example (in a j-job, but it can be removed from dev drivers too):

https://github.com/NOAA-EMC/EVS/blob/2cc2a70daa127995115056dacdec9fb864ae82c3/jobs/aqm/plots/JEVS_AQM_PLOTS#L37

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Please change geoffrey.manikin@noaa.gov to alicia.bentley@noaa.gov in your maillists! Thanks!

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Please change emc.vpppg back to $USER and KEEPDATA back to NO on the following lines in [dev/drivers/scripts/aqm/stats/jevs_aqm_grid2obs_stats.sh](https://github.com/NOAA-EMC/EVS/pull/232/files#diff-a7e20c0e49c5a24f8317b3310c619ed71a7abfb266a2ef245cbc41e115eb1881). Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @PerryShafran-NOAA , I did a quick initial code review and everything looks good. What I know less, and I apologize for getting it lost in the zillions of emails, but does this PR incorporate the change(s) @Ho-ChunHuang-NOAA needed?

**Comment by PerryShafran-NOAA** [comment]
There was a lot of back and forth between one of the things he suggested, but the upshot of that is we are good for now in terms of his needs.  

I'm not sure you actually ran any tests, but it would be good to have tests of everything as well.

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks, @PerryShafran-NOAA ! I have not run any tests yet. I will begin now and report when I have the prep, stats, and plots completed.

**Comment by ShelleyMelchior-NOAA** [comment]
The aqm prep process has completed. Please review
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm/atmos/atmos.20230910/aqm/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_prep_00.o81792925`

**Comment by PerryShafran-NOAA** [comment]
2 things:

This worked fine; there is SENDCOM working and there are files in the prep directory.

But:

- I need to define FIXevs in the J-job
- There is a MET_bin_exec definition in there as well.

Will need to fix.

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks, @PerryShafran-NOAA . Prep for aqm is re-running.

Stats for aqm was also run and it has problems I can't readily identify.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o81794704`

**Comment by PerryShafran-NOAA** [comment]
It's actually the same error.  Instead of FIXevs, I have FIXvevs for whatever reason.  That was a problem in prep too but nothing in prep used anything from fix I don't think.  Anyway, please try again.

**Comment by ShelleyMelchior-NOAA** [comment]
Well good! Removing FIXevs from drivers revealed a typo but in the J-job!

**Comment by ShelleyMelchior-NOAA** [comment]
prep and stats were re-run and look good!

`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/`
COMOUT=` /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/prep/aqm/atmos/atmos.20230910/aqm`

`/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/atmos.20230910/aqm/grid2obs`

**Comment by PerryShafran-NOAA** [comment]
Another whoops.  It turns out that /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/aqm.20230910 is a file, not a directory.

Please remove that file aqm.20230910.  And then re-run.  I just made the correction to this one.

**Comment by ShelleyMelchior-NOAA** [comment]
I re-ran stats after pulling in your update and removing the bad file. Have a look.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/aqm/`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_stats.o81802260`

**Comment by PerryShafran-NOAA** [comment]
Everything looks good and as expected.  I think that the stats are good to go.

Perry

File: dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh:1]: Please change line 82 to remove "emc.vpppg" and use $USER instead.
`export COMINaqm=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v1.0/stats/aqm/aqm`

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/aqm/plots/jevs_aqm_grid2obs_plots.sh:1]: Done, thanks for checking!

**Comment by ShelleyMelchior-NOAA** [comment]
The plots processing has completed.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230911`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o81861221`

**Comment by PerryShafran-NOAA** [comment]
Hmmm, the plot file is empty.  2 issues:

1) I think that the stats files from dogwood did not transfer.  Let me transfer them.
2) FIXevs:  I think I have the same FIXvevs error in the plot job.  Let me correct that.

Perry

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Thanks for finding these things, Perry! Please let us know when the stats transfer is complete!

**Comment by ShelleyMelchior-NOAA** [comment]
I re-ran the plots processing and it has finished and the tar is not 0 size! I will take the small victories. How does the content look, @PerryShafran-NOAA ?
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230911`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o81882943`

**Comment by PerryShafran-NOAA** [comment]
Darn; I have an error from Marcel's fix in there that is causing many missing plots.  I can fix that; hold please.

**Comment by PerryShafran-NOAA** [comment]
OK, please re-run the plot job.  I made a fix in there.

**Comment by ShelleyMelchior-NOAA** [comment]
The plots re-run completed. Please review.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/atmos.20230911`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o81890392`

**Comment by PerryShafran-NOAA** [comment]
Darn.  This fix needs fixing.  Hold on.

**Comment by PerryShafran-NOAA** [comment]
Try one more time, and I will also try to get Marcel's help.

Perry

**Comment by PerryShafran-NOAA** [comment]
Did you get the latest latest update?  Based on the error I see, you didn't.  Please bring in the changes and try one more time.

**Comment by ShelleyMelchior-NOAA** [comment]
This is the file in my checkout of your branch:
`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/githubwkspc/forks/PS/EVS/ush/aqm/lead_average.py`
I looked at line 1007 and I think I have the changes.

**Comment by PerryShafran-NOAA** [comment]
Yes, you now have them. Please run plots again

**Comment by PerryShafran-NOAA** [comment]
I just checked your latest .o output file.  The fhrmean plots are back!  There were so many issues dealing with the proper parenthesis/brackets/etc of the statement Marcel helped me put them in, but now that seems to be correct.

One thing I'd like to see.  Can you show me the plot of the file evs.aqm.fbar_obar.pm25_l1.last31days.fhrmean_init12z.buk_conus.png to see if the plot issue has been resolved?  I'm having trouble with XLaunch on my computer I think.   I think that I showed Ho-Chun that plot earlier and he said it looked better.  Once I see that, then we can merge this one.

Thanks!

**Comment by ShelleyMelchior-NOAA** [comment]
Hi @PerryShafran-NOAA , I am back to my desk. Yes, the plot finished.
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/aqm/
atmos.20230911`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o81893992`
I expanded the tar file on emcrzdm: https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/aqm/
I think this is the exact plot you specified:
https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/aqm/evs.aqm.fbar_obar.pm25_l1.last31days.fhrmean_init12z.buk_conus.png

I don't see the logos ... did I forget to make the fix/ dir? I did! Dang it! Sigh. Let me run again ...

**Comment by ShelleyMelchior-NOAA** [comment]
plots re-run in progress ... again.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_aqm_grid2obs_plots.o81920079`
DATA=`/lfs/h2/emc/stmp/shelley.melchior/evs_test/prod/tmp/jevs_aqm_grid2obs_plots.81920079.cbqs01`

**Comment by PerryShafran-NOAA** [comment]
Please show me the same plot once the run completes, just to see the logos.  The obs count on the top of the plot look fine, though.

**Comment by AliciaBentley-NOAA** [comment]
Yessss. Hopefully we can get this merged tonight!

___________________________________
*Alicia M. Bentley, Ph.D.*
Global Verification Project Lead
Model Evaluation Group (MEG) <https://www.emc.ncep.noaa.gov/users/meg/home/>
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by ShelleyMelchior-NOAA** [comment]
Logos! https://ftp.emc.ncep.noaa.gov/users/smelchior/EVS/aqm/evs.aqm.fbar_obar.pm25_l1.last31days.fhrmean_init12z.buk_conus.png

**Comment by PerryShafran-NOAA** [comment]
Whew!  Everything's perfect.  I think we're good to merge.

**Comment by AliciaBentley-NOAA** [comment]
🎉🎉🎉🎉🎉🎉🎉🎉🎉🎉

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
This code is well tested and approved for merge.

**Pull request merged**